### DISCUSSION TITLE
Allow arbitrary connectors with arbitrary widgets

### DISCUSSION THREAD

**Discussion by fubhy** [family/connectkit#251] [General]
I'd like to register a custom connector for impersonating (without requiring a browser extension or sth. like https://www.impersonator.xyz/).

For that, I'd like to provide a custom form widget (for entering the address of the wallet & network to impersoate).

In general I think it would be a good idea and improve the code quality & composability if connectors would register their connection widget & logic instead of having that handled centrally with a conditional to render the appropriate qr code:

https://github.com/family/connectkit/blob/c77dc936ae35834a6c9a0628c4bd26cdff676197/packages/connectkit/src/components/ConnectModal/ConnectWithQRCode.tsx#L42

**Comment by lochie** [comment]
I like this idea. What would the custom connector look like from a code perspective? Just an input with a wallet address?

**Reply by fubhy** [reply]
An input for the wallet address and a dropdown for the network.

And optionally a custom RPC node that could be used (if unlocked) to actually simulate transactions.

**Reply by lochie** [reply]
Could work! fwiw I've seen someone [fork ConnectKit](https://github.com/janjakubnanista/connectkitx) to include MagicLink which opens a modal ontop of the ConnectKit modal with form inputs. Perhaps that's an easier setup without having to include custom widget injection?

**Reply by grothem** [reply]
We have a similar use case that this would be really helpful for. We just need an input for the user to put in a wallet address.
